switchboard/base.py

```diff
@@ -6,77 +6,91 @@
 :license: Apache License 2.0, see LICENSE for more details.
 """
 
-import time
-import logging
 import threading
 
-from .models import MongoModel
-from .signals import request_finished
-from .settings import settings
 
-log = logging.getLogger(__name__)
+class ModelDict(threading.local):
+    """
+    Dictionary-style access to model data.
```

Comment:
Great docstring, I'd also add an explanation about why the data here needs to be local to the thread.
Also, what is a Model? Could we define it as an inteface? For instance, it must be an object that defines a set of methods: get(str) -> str etc.